### ISSUE TITLE
Add support for renaming to purs ide

### ISSUE THREAD

**Issue by alextes** [purescript/purescript#3732]
I'm not sure what the correct term is for name, label, which is used to reference other things in our code, maybe 'symbol'? I know this already has meaning in PureScript on the type level but I hope the definition is clear. Anything that has a name made up by the dev.

Would it be possible to add make purs ide capable of renaming labels for us? Like a smart find and replace that works project-wide. Language servers call this rename.

**Comment by kritzcreek** [comment]
We already have support for finding identifier usages, which is closely related to being able to rename something project wide.

The main difference is that for a rename operation you need to be more precise and make sure you find _every_ reference. The code that finds usages is here: https://github.com/purescript/purescript/blob/3189d931471e88368f3d78b4435a9a96dfb9f312/src/Language/PureScript/Ide/Usage.hs, but as I comment here: https://github.com/purescript/purescript/blob/3189d931471e88368f3d78b4435a9a96dfb9f312/src/Language/PureScript/Ide/Usage.hs#L82-L83
there are some cases that need to be handled more precisely to allow for a renaming.

**Comment by whitecolor** [comment]
That PS doesn't support the simplest refactoring operations (renaming types, functions, etc) in the editor for me was almost revelation. I didn't think that functional programmers are so brutal 👊.

**Comment by whitecolor** [comment]
@kritzcreek is there a milestone when those capabilities are going to be added?

**Comment by kritzcreek** [comment]
@whitecolor No. As I said I don't plan on implementing it myself, but I'd invest the time to review a contribution.
